src/shared/components/CTag.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+  <span :class="['tag', variantClass]">
+    <span class="tag__title">{{ title }}</span>
+
+    <img v-if="icon" :src="icon" class="tag__icon" />
+
+    <button v-if="removable" @click.stop="onRemove">
+      <img :src="XCircleIcon" className="tag__remove" />
+    </button>
+  </span>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue';
+import { XCircleIcon } from '../assets/icons';
+
+type Variant = 'default' | 'outline' | 'error' | 'alert';
+
+const props = withDefaults(
+  defineProps<{
+    variant?: Variant;
+    icon?: string;
+    title: string;
+    removable?: boolean;
+  }>(),
+  {
+    variant: 'default',
+    removable: false,
+  }
+);
+
+const variantClass = computed(() => `tag--${props.variant}`);
+
+const emit = defineEmits<{
+  (e: 'remove'): void;
+}>();
+
+const onRemove = () => {
+  emit('remove');
+};
```

Comment:
passa como segundo parâmetro do define emit emit o o nome do valor e seu tipo (id: number). Já a emissão em si, passa o id da tag, assim no componente pai onde uma função onRemove for usado, passa-se apenas a função sem necessidade de passar um parâmtro como está sendo feito atualmente (`@remove=onRemove(tag.id)`).

No geral, essa abordagem é mais adequada.

```suggestion
const emit = defineEmits<{
  (e: 'remove', id: number): void;
}>();

const onRemove = () => {
  emit('remove', props.id);
};
```